torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3376,6 +3377,127 @@ def test_ddp_hook_parity_powerSGD(self):
                 )
                 self._test_ddp_hook_parity(state=powersgd_state, hook=powerSGD.powerSGD_hook)
 
+        def _prepare_module(
+            self,
+            process_group,
+            global_batch_size,
+            gradient_as_bucket_view=False,
+        ):
+            model = Net()
+            ddp_model = DistributedDataParallel(
+                copy.deepcopy(model),
+                process_group=process_group,
+                bucket_cap_mb=0.001,
+                gradient_as_bucket_view=gradient_as_bucket_view,
+            )
+            input = torch.randn(global_batch_size, 2)
+            target = torch.randn(global_batch_size, 4)
+            return model, ddp_model, input, target
+
+        def _test_accumulate_gradients_no_sync(
```

Comment:
Looks like this function is copied from `test_c10d_nccl.py`? If yes, let's dedup.

Reply:
migrated comm hook NCCL tests from test_c10d_nccl.py to here